rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -15,16 +15,97 @@
 #ifndef RMW_CONNEXT_SHARED_CPP__WAIT_HPP_
 #define RMW_CONNEXT_SHARED_CPP__WAIT_HPP_
 
+#include <unordered_set>
+
 #include "ndds_include.hpp"
 
 #include "rmw/error_handling.h"
 #include "rmw/impl/cpp/macros.hpp"
 #include "rmw/types.h"
 
 #include "rmw_connext_shared_cpp/condition_error.hpp"
+#include "rmw_connext_shared_cpp/event_converter.hpp"
 #include "rmw_connext_shared_cpp/types.hpp"
 #include "rmw_connext_shared_cpp/visibility_control.h"
 
+
+rmw_ret_t __gather_event_conditions(
+  rmw_events_t * events,
+  std::unordered_set<DDS::StatusCondition *> & status_conditions)
+{
+  if (events) {
+    // gather all status conditions and masks
+    for (size_t i = 0; i < events->event_count; ++i) {
+      rmw_event_t * current_event = events->events[i];
+      DDSEntity * dds_entity =
+        static_cast<DDSEntity *>(current_event->data);
+      if (!dds_entity) {
+        RMW_SET_ERROR_MSG("Event handle is null");
+        return RMW_RET_ERROR;
+      }
+      DDS::StatusCondition * status_condition = dds_entity->get_statuscondition();
+      if (!status_condition) {
+        RMW_SET_ERROR_MSG("status condition handle is null");
+        return RMW_RET_ERROR;
+      }
+
+      DDS_StatusMask new_mask = status_condition->get_enabled_statuses() |
+        get_mask_from_rmw(current_event->event_type);
```

Comment:
Would it be better to return RMW_RET_EVENT_UNSUPPORTED if the event isn't in the get_mask_from_rmw map?

Reply:
Resolving as we are returning RMW_RET_ERROR in other cases above as discussed.